creator/lib/app.dart

```diff
@@ -1,58 +1,142 @@
+import 'dart:ui';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:creator/common/color.dart';
-import 'package:creator/screens/salon_applying_screen.dart';
+import 'package:creator/firebase/database.dart';
+import 'package:creator/screens/home_screen.dart';
+import 'package:creator/screens/login_screen.dart';
+import 'package:creator/screens/reason_for_creating_the_salon_screen.dart';
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
 
+bool userLoggedIn = false;
+
 class KSCreatorApp extends StatelessWidget {
   const KSCreatorApp({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      home: SalonApplyingScreen(),
-      theme: ThemeData(
-        elevatedButtonTheme: ElevatedButtonThemeData(
-          style: ButtonStyle(
-            backgroundColor: MaterialStateProperty.resolveWith<Color>(
-              (Set<MaterialState> states) {
-                if (states.contains(MaterialState.disabled)) {
-                  return Colors.white38;
-                }
-                return Colors.blue;
-              },
-            ),
-          ),
-        ),
-        scaffoldBackgroundColor: primaryColor,
-        appBarTheme: AppBarTheme(
-          backgroundColor: Colors.white,
-          textTheme: TextTheme(
-            headline1: TextStyle(
-              color: primaryColor,
-            ),
+      theme: _buildThemeData(),
+      routes: {
+        '/salon_applying_screen': (context) =>
+            ReasonsForCreatingTheSalonScreen(),
+        '/home': (context) => HomeScreen(),
+      },
+      home: StreamBuilder(
+        stream: FirebaseAuth.instance.authStateChanges(),
+        builder: (_, snapshot) {
+          // final user = snapshot.data as dynamic;
+
+          if (snapshot.hasData) {
+            return _buildHomePage();
+          } else {
+            return LoginScreen();
+          }
+
+          // if (snapshot.hasData &&
+          //     FirebaseFirestore.instance
+          //             .collection(DbHandler.usersColletion)
+          //             .doc(user.email) !=
+          //         null) {
+          //   return _buildHomePage();
+          // }
+          // return LoginScreen();
+        },
+      ),
+    );
+  }
+
+  ThemeData _buildThemeData() {
```

Comment:
Please use this function if possible: https://github.com/Gospel-Crowd/salon-app/blob/main/salon_creator/lib/app.dart#L53

Reply:
I used this function.